### PR TITLE
feat: Wasi host callback

### PR DESCRIPTION
**Pull request by flavio**
Allow Wasi policies to make host calls. This makes possible to use host capabilities like Sigstore integration, but also to make queries against the Kubernetes API.

This makes possible to write Wasi policies that are context aware.

Some notes about this PR:

- The integration tests require a test policy that cannot be published yet. We require a new patched version of kwctl to release it
- The Go SDK has to be updated to be able to handle this change - a dedicated PR is going to be made
- **Most important:** pre-existing Wasi policies are not broken by this change

### CODE REVIEW

**Review by fabriziosestito — APPROVED** [review]
LGTM

File: src/runtimes/callback.rs [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+    mut rx: Receiver<Result<CallbackResponse>>,
+    eval_ctx: &EvaluationContext,
+) -> Result<Vec<u8>, Box<dyn std::error::Error + Send + Sync>> {
+    let cb_channel: mpsc::Sender<CallbackRequest> =
+        if let Some(c) = eval_ctx.callback_channel.clone() {
+            Ok(c)
+        } else {
+            error!(
+                policy_id,
+                binding, operation, "Cannot process waPC request: callback channel not provided"
```
Comment (viccuad) [inline src/runtimes/callback.rs:384]:
```suggestion
                binding, operation, "Cannot process request: callback channel not provided"
```

These requests may be waPC or Wasi (same for other err messages).

File: src/runtimes/wasi_cli/wasi_pipe.rs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    io::{Read, Seek, Write},
+};
+
+// Code based on https://github.com/lapce/lapce/blob/f7d2f4ba863e3a00c9bcb0f3ac1896fd446e7d4c/lapce-proxy/src/plugin/wasi.rs#L45
+#[derive(Default)]
+pub(crate) struct WasiPipe {
+    buffer: VecDeque<u8>,
+}
+
+impl WasiPipe {
```
[the remark this answers is not in the document]
Comment (flavio) [reply src/runtimes/wasi_cli/wasi_pipe.rs:17]: Done

**Review by viccuad — APPROVED** [review]
LGTM!

File: tests/integration_test.rs [hunk 3]
```diff
@@ -206,11 +206,18 @@ async fn test_policy_evaluator(
 }
 
 #[rstest]
+// TODO: uncomment once the actual policy is published
```
Comment (fabriziosestito) [inline tests/integration_test.rs:209]: reminder to uncomment this

Comment (flavio) [reply tests/integration_test.rs]: Fixed, the code is now testing the actual policy

**Comment by flavio** [comment]
rebased against the latest changes done inside of the main branch, plus consume the real go-wasi-context-aware test policy inside of the integration tests

**Pull request merged**